### PR TITLE
feat: Introduces connection event for custom properties.

### PR DESCRIPTION
**Pull request by damencho**
Used for shard and region values coming from the backend.

### CODE REVIEW

**Review by jallamsetty1 — APPROVED** [review]

File: modules/xmpp/xmpp.js [hunk 1]
```diff
@@ -1110,5 +1110,16 @@ export default class XMPP extends Listenable {
         }
 
         this.sendDeploymentInfo = false;
+
+        const { region, shard } = aprops;
+
+        if (region || shard) {
+            // avoids sending empty values
+            this.eventEmitter.emit(JitsiConnectionEvents.PROPERTIES_UPDATED, JSON.parse(JSON.stringify({
```
Comment (saghul) [inline modules/xmpp/xmpp.js:1118]: Suggestion: use lodash clone or structuredClone.

**Review by saghul — APPROVED** [review]

File: modules/xmpp/xmpp.js (hunk 1 above)
Comment (damencho) [reply modules/xmpp/xmpp.js:1118]: That will drop the keys with undefined values?

Comment (saghul) [reply modules/xmpp/xmpp.js:1118]: No. Do you want to?

Comment (damencho) [reply modules/xmpp/xmpp.js:1118]: Yes. That was a quick hack to avoid if this and if that to not add the keys.

Comment (saghul) [reply modules/xmpp/xmpp.js:1118]: Ah then go ahead!

**Pull request merged**